clientdb/batch_snapshot.go

```diff
@@ -465,6 +495,40 @@ func deserializeLocalBatchSnapshot(r io.Reader) (*LocalBatchSnapshot, error) {
 		b.MatchedOrders[nonce] = append(b.MatchedOrders[nonce], m)
 	}
 
+	var (
+		numPrices uint32
+		duration  uint32
+		price     order.FixedRatePremium
+	)
+
+	// Older batches just had one clearing price. So if there is no
+	// additional data to read here, we can assume we have such an order.
+	err = ReadElements(r, &numPrices)
+	switch err {
+	// Successful read, continue with reading the prices.
+	case nil:
+
+	// If it wasn't found in the database, then we'll assume the default
+	// value.
+	case io.EOF, io.ErrUnexpectedEOF:
+		b.ClearingPrices[order.LegacyLeaseDurationBucket] = clearingPrice
```

Comment:
Could avoid the use of `EOF` markers by instead checking if the clearing price is non-zero signaling an old batch.

Reply:
Good idea! I thought it might screw with our unit tests if we always write a zero price. But it turns out this works well.